riemann/serializer.js

```diff
@@ -2,50 +2,74 @@
    and cache it in memory. */
 var riemannSchema;
 if (!riemannSchema) {
-  var Schema    = require('node-protobuf');
-  var readFile  = require('fs').readFileSync;
-  riemannSchema = new Schema(readFile(__dirname+'/proto/proto.desc'));
+  var protobuf = require('protobufjs')
+
+  protobuf.load(__dirname + '/proto/proto.proto', function (err, root) {
+    if (err) throw err
+
+    // Pull the message type out.
+    riemannSchema = root
+  })
 }
 
-function _serialize(type, value) {
-  return riemannSchema.serialize(value, type);
+function _serialize (type, value) {
+  var messageType = riemannSchema.lookupType(type)
+
+  // https://www.npmjs.com/package/protobufjs#valid-message
+  var errorString = messageType.verify(value)
+  var message
+
+  // Using create is faster, so only fall back to fromObject in worst case.
+  if (typeof errorString === 'string') {
```

Comment:
I'm not sure how to interpret this? why is the return type of verify significant?

Reply:
protobufjs signals an invalid (non-conformant) object by populating the return value of the verify call with a string of the error message.  If there are no verification errors (such as trying to shove a float value into an int slot in the proto file) it will instead return NULL, which indicates no errors.